src/libsyntax/feature_gate.rs

```diff
@@ -395,12 +403,15 @@ impl<'a> Context<'a> {
                               "attributes of the form `#[derive_*]` are reserved \
                                for the compiler");
         } else {
-            self.gate_feature("custom_attribute", attr.span,
-                       &format!("The attribute `{}` is currently \
-                                unknown to the compiler and \
-                                may have meaning \
-                                added to it in the future",
-                                name));
+            // Only do the custom attribute lint post-expansion
```

Comment:
`feature_gate::check_crate_macros` happens before the registry is loaded (so `plugin_attributes` is empty and it will error on attributes even if they get registered later). I guess I should say "Only do the custom attribute lint during regular feature gate checking"